src/js/components/TextInput/TextInput.js

```diff
@@ -115,6 +115,26 @@ const TextInput = forwardRef(
     const [focus, setFocus] = useState();
     const [showDrop, setShowDrop] = useState();
 
+    const handleSuggestionSelect = useRef();
```

Comment:
The way these are being used seems like they should be using `useMemo` instead of `useRef + useEffect`.

Reply:
I'm a little confused about this. My understanding of useMemo is to optimize the performance of complex functions. How can I use it to replace useRef and useEffect?